Migrate Navigable toolbar test to Playwright

pooja-muchandikar opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
## What?
<!-- In a few words, what is the PR actually doing? -->
Part of https://github.com/WordPress/gutenberg/issues/38851.
Migrate[ navigable-toolbar.test.js ](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/navigable-toolbar.test.js) to its Playwright version.
## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Part of https://github.com/WordPress/gutenberg/issues/38851.
## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->
Run` npm run test:e2e:playwright test/e2e/specs/editor/various/navigable-toolbar.spec.js`

pooja-muchandikar committed:
Migrate Navigable toolbar test to Playwright

pooja-muchandikar commented:
Hi @Mamaduka @jeryj 

I have created another PR with feedbacks being addressed, but somehow the CI is failing. Test case is passing in my local though. Please share your suggestions on same.

![image](https://github.com/WordPress/gutenberg/assets/41000648/c7628c53-1144-476e-8d1a-e08b1d926e3b)

jeryj commented on test/e2e/specs/editor/various/navigable-toolbar.spec.js:
```diff
@@ -0,0 +1,47 @@
+
+	test.describe( 'Contextual Toolbar', () => {
+		test( 'should not scroll page', async ( { page, pageUtils } ) => {
+			while (
+				await page.evaluate( () => {
+					const scrollable = window.wp.dom.getScrollContainer(
+						document.activeElement
+					);
+					return ! scrollable || scrollable.scrollTop === 0;
+				} )
+			) {
+				await page.keyboard.press( 'Enter' );
+			}
```
I'm not sure why yet, but when I run this test locally and watch it in the browser, it gets stuck in a cycle of pressing Enter over and over without moving on to the code after it.

jeryj replied:
I think something got messed up during a copy/paste or search/replace. I copy/pasted the old test from `[navigable-toolbar.test](https://raw.githubusercontent.com/WordPress/gutenberg/6c0f4a50324b7f09b4c13c8ce2b974b1e2243bd1/packages/e2e-tests/specs/editor/various/navigable-toolbar.test.js)` and replaced the `expect( await isInBlockToolbar() ).toBe( true );` part, and the test passes. 

Could you try copy/pasting it again and see if it works for you?

jeryj commented on packages/e2e-tests/specs/editor/various/navigable-toolbar.test.js:
```diff
@@ -1,104 +0,0 @@
-		it( 'should not scroll page', async () => {
-			while (
-				await page.evaluate( () => {
-					const { activeElement } =
-						document.activeElement?.contentDocument ?? document;
```
I think this part needed to get moved over to the playwright test here and in `scrollTopBefore`

pooja-muchandikar committed:
Address feedbacks

pooja-muchandikar commented:
Hi @jeryj,

I have addressed the feedbacks and the CI also passed successfully!

Do let me know if there are any further suggestions..

Thanks!

pooja-muchandikar commented:
Hi @jeryj, 

I have addressed the feedbacks and pushed the code and CI had passed, but seems now the CI got executed automatically and it failed again. Though it is not related to the test case I worked on. 

![image](https://github.com/WordPress/gutenberg/assets/41000648/468d273e-b877-4ae5-af32-f615590fcb5d)

jeryj reviewed (approved):
Looks great! Thanks for working on this! 🚢
